### PR TITLE
Feat(veto-frontend): VN-104: adds a main layout and the logic for navigation from side menu

### PR DESCRIPTION
**Pull request by gbibeaul**
**Description**
---

- [x] adds a component taking care of rendering connected routes 
- [x] adds ability to navigate from the side menu
- [x] improves some typing where I was working

This is some preparation work for the styling to come. This is by no means a complete version of the layout but does provide the ground work to make the rest of it and apply it to all our routes

### CODE REVIEW

File: packages/veto-frontend/src/layouts/MainLayout/MainLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export type MainLayoutLabel = 'Dashboard' | 'Wallets' | 'History' | 'Explorer' | 'Monitoring'
+export type MainLayoutRoute = '/dashboard' | '/wallets' | '/history' | '/explorer' | '/monitoring'
+
+type Props = {
+  path: string
+}
+
+const getActiveMenu = (): MainLayoutRoute => {
+  const currentPathName: string = window.location.pathname
+  if (currentPathName !== '/dashboard' || '/wallets' || '/history' || '/explorer' || '/monitoring') {
```
Comment (drawm) [inline packages/veto-frontend/src/layouts/MainLayout/MainLayout.tsx]: Are you working with Python theses days :P  
That patter does not work in JS, you can do something similar with `Array.prototype.has()`

```typescript
const pathWhitelist = ['/dashboard', '/wallets', '/history', '/explorer', '/monitoring'];
if(pathWhitelist.has(currentPathName)){
  return currentPathName;
} else {
  return '/dashboard';
}


```

File: packages/veto-frontend/src/components/molecules/SideMenu/SideMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React, { FunctionComponent } from 'react'
+import { Flex } from '@veriphi/veto-ui'
+import { MainLayoutLabel, MainLayoutRoute } from '../../../layouts/MainLayout/MainLayout'
```
Comment (nk1tz) [inline packages/veto-frontend/src/components/molecules/SideMenu/SideMenu.tsx]: We have absolute imports working in `veto-frontend` package with `baseUrl='src'` so you should be able to `import from 'layouts/MainLayout`

File: packages/veto-frontend/src/layouts/MainLayout/MainLayout.tsx (hunk 1 above)
Comment (nk1tz) [reply packages/veto-frontend/src/layouts/MainLayout/MainLayout.tsx]: I think he mentioned working with Python recently actually! :P

But unfortunately `.has` is a Map prototype. 
In your example @drawm we'd have to use  the equivalent Array prototype `.includes` 

```javascript

if(
   ['/dashboard', '/wallets', '/history', '/explorer', '/monitoring'].includes(currentPathName)
){
   return currentPathName;
} else {
   return '/dashboard';
}

```

Comment (drawm) [reply packages/veto-frontend/src/layouts/MainLayout/MainLayout.tsx]: Oh right! I always mix up the two :P `.includes` is the way to go !

**Comment by gbibeaul** [comment]
closing in favor of #27 since new folder structure made conflicts complexe to solve

**Comment by drawm** [comment]
See guillaumes message above
> closing in favor of #27 since new folder structure made conflicts complexe to solve

**Pull request closed without merging**